WalletWasabi.Fluent/ViewModels/Dialog/DialogViewModelBase`1.cs

```diff
@@ -0,0 +1,52 @@
+using System;
+using System.Collections.Generic;
+using System.Reactive.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace WalletWasabi.Fluent.ViewModels.Dialog
+{
+	/// <summary>
+	/// Base ViewModel class for Dialogs that returns a value back.
+	/// </summary>
+	/// <typeparam name="TResult">The type of the value to be returned when the dialog is finished.</typeparam>
+	public abstract class DialogViewModelBase<TResult> : DialogViewModelBase
+	{
+		private TaskCompletionSource<TResult>? _tcs { get; set; } 
+
+		public DialogViewModelBase(MainViewModel mainViewModel) : base(mainViewModel)
+		{
+		}
+
+		/// <summary>
+		/// Method to be called when the dialog is finished
+		/// and ready to pass a value back to the caller.
+		/// </summary>
+		/// <param name="value">The return value of the dialog</param>
+		public void DialogFinished(TResult value)
+		{
+			_tcs.SetResult(value);
+			_tcs = null;
+		}
+
+		/// <summary>
+		/// Shows the dialog.
+		/// </summary>
+		/// <returns>The value to be returned when the dialog is finished.</returns>
+		public Task<TResult> ShowDialogAsync()
+		{
+			_tcs = new TaskCompletionSource<TResult>();
```

Comment:
what happens if _tcs is not already null here?

best throw an exception or somethung... `Dialog Already Opened`